### PR TITLE
US State Map

### PR DESCRIPTION
**Pull request by hughess**
### Description
Our first map! This PR includes a new component `USMap` which is a choropleth map of the US states. The component takes a data object, `state` column, and `value` column.

CleanShot 2023-01-25 at 14 30 42

### Props for `USMap`
- `data` - query result
- `state` - column to be used as the name for each state
- `value` - column to be used as the value determining the colour of each state
- `abbreviations` - false | true - if true, map will look for two letter abbreviations rather than full names. Default is false
- `colorScale` - see section below
- `min` - minimum value for the colour scale. Anything below the minimum will be shown in the same colour as the min value
- `max` - maximum value for the colour scale. Anything above the maximum will be shown in the same colour as the max value
- `title`
- `subtitle`

### Colour Palettes
The map is set up to receive a `colorScale` prop which can be `blue`, `green`, `red`, or `bluegreen`. Default colour scale is `blue`.
CleanShot 2023-01-25 at 14 42 03@2x
CleanShot 2023-01-25 at 14 41 20@2x
CleanShot 2023-01-25 at 14 41 30@2x
CleanShot 2023-01-25 at 14 41 10@2x

### Open Issues
- This map requires a geo JSON file to be loaded into every project, unless we decide to host the file. The file is 62 KB
- Spacing on mobile is an issue - the container is much too tall at the moment
- There are likely some reactivity issues with the map component - to be tested a bit more before merging
- This map excludes Alaska, Hawaii, and Puerto Rico - we can easily add these back in, but they make it more difficult to space the chart in the article, so they'll need some extra consideration
- There is significant duplication of code in this PR between our chart library and the components required to make this map work. This should be collapsed and simplified in the future

### Opportunities
- The ECharts map chart type is quite powerful. It's easy to create geoJSON files (e.g., from https://geojson-maps.ash.ms) and plug them into this. We can extend this in a couple of ways:
   - We can source other geoJSON files and offer them as built-in map components in the same way as US States
   - We can open up the components to accept a custom geoJSON that someone puts into their project. Spacing of the map on the page and on mobile would be the challenge here, but in theory this would allow you to create any choropleth you can think of

### Checklist
- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

**Comment by hughess** [comment]
Mobile spacing on this one has been a bit tricky, but with a combination of echarts' media queries, CSS media queries, and some logic to handle title and subtitle spacing, this works pretty well in most cases:

CleanShot 2023-01-26 at 10 08 03

File: sites/example-project/src/components/viz/USMap.svelte [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    abbreviations = (abbreviations === "true" || abbreviations === true);
+
+    let nameProperty = abbreviations ? "abbrev" : "name"
+
+    let columnSummary
+    try {
+      checkInputs(data, [state, value]);
+      columnSummary = getColumnSummary(data);
+    } catch(e) {
+      error = e.message;
```
Comment (Winterhart) [inline sites/example-project/src/components/viz/USMap.svelte:142]: I guess this is another place to add the `strict build` error throwing 😅

File: sites/example-project/src/components/viz/USMap.svelte [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        '#e5f5f9',
+        '#ccece6',
+        '#99d8c9',
+        '#66c2a4',
+        '#41ae76',
+        '#238b45',
+        '#006d2c',
+        '#00441b'
+
+        // '#c6f5c4',
```
Comment (Winterhart) [inline sites/example-project/src/components/viz/USMap.svelte:49]: Should we remove those commented out css colours?

**Review by Winterhart — COMMENTED** [review]
I have one or two questions... it looks amazing 👍🏻

File: sites/example-project/src/components/modules/echartsMap.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import * as echarts from 'echarts';
```
[the remark this answers is not in the document]
Comment (hughess) [reply sites/example-project/src/components/modules/echartsMap.js:1]: Good point - I think we can also do this in the charting library!

**Review by Winterhart — APPROVED** [review]
Cool it's a great start for the map feature 🚀

**Comment by archiewood** [comment]
This may not be related to the USMap (possibly the data objects) but the console is throwing an error when I show the queries on the page:
CleanShot 2023-01-26 at 20 56 15

<details>
<summary><h2> Full Error Trace</h2> </summary>
```
Uncaught (in promise) TypeError: i is undefined
    ue https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/getColumnSummary-fbb006f4.js:1
    update https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/QueryViewer-269da9db.js:1
    Qe https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    _t https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/QueryViewer-269da9db.js:1
    Fe https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/QueryViewer-269da9db.js:3
    Ye https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/QueryViewer-269da9db.js:3
    p https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/QueryViewer-269da9db.js:3
    Ut https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    it https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    promise callback*wt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    Vt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    ctx https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    Kt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    r https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-31f3c46c.js:1
    b https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-31f3c46c.js:1
    c https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    _e https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:1
    m https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    m https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    m https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    Kt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    m https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    Kt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    m https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    m https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/pages/__layout.svelte-787eedc7.js:1
    Kt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    m https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/start-9c60de77.js:1
    Kt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    Qe https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/index-c2500592.js:4
    Et https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/start-9c60de77.js:1
    _init https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/start-9c60de77.js:1
    start https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/start-9c60de77.js:1
    kt https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/start-9c60de77.js:1
    <anonymous> https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/maps:36
[getColumnSummary-fbb006f4.js:1:6299](https://evidence-development-workspace-d1f2fatwf-evidence-dev.vercel.app/_app/chunks/getColumnSummary-fbb006f4.js)
```

</details>

**Comment by hughess** [comment]
@archiewood I think this was caused by an issue I fixed yesterday - the commits I just pushed should fix it. I had accidentally taken a shallow copy of the data object and manipulated it, so I think that was breaking the query viewer

**Pull request merged**